src/lib/list/list.ts

```diff
@@ -6,28 +6,50 @@ import {
     QueryList,
     Directive,
     ElementRef,
+    Inject,
+    Input,
+    OpaqueToken,
     Renderer,
     AfterContentInit,
     NgModule,
     ModuleWithProviders,
 } from '@angular/core';
-import {MdLine, MdLineSetter, MdLineModule} from '../core';
+import {MdLine, MdLineSetter, MdLineModule, MdRippleModule} from '../core';
 
 @Directive({
   selector: 'md-divider'
 })
 export class MdListDivider {}
 
+export enum ListType {
+  NORMAL_LIST,
+  NAV_LIST,
+}
+
+/**
+ * Token used to inject the list type into child MdListItem components so they can know whether
+ * they're in a nav list (and thus should use an MdRipple).
+ */
+const LIST_TYPE_TOKEN = new OpaqueToken('is_nav_list');
```

Comment:
@dozingcat  Ran into a google presubmit problem: this token has to be exported (for AoT)